__tests__/setup/postActivity.js

```diff
@@ -0,0 +1,23 @@
+import updateIn from 'simple-update-in';
+
+import waitForActivity from './waitForActivity';
+import waitForObservable from './waitForObservable';
+
+const DEFAULT_USER_ID = 'u-12345';
+
+export default async function postActivity(directLine, activity) {
+  // We need to use channelData.clientActivityId because postActivity could come later than the activity$ observable.
+  // Thus, when we receive the activity ID for the "just posted" activity, it might be already too late.
+
+  const targetClientActivityId = Math.random().toString(36).substr(2);
+
+  activity = updateIn(activity, ['from', 'id'], userId => userId || DEFAULT_USER_ID);
+  activity = updateIn(activity, ['channelData', 'clientActivityId'], () => targetClientActivityId);
+
+  const [activityId] = await Promise.all([
```

Comment:
Is this just returning a boolean value?

Reply:
No, this will return the `activityId`, although we did not use it for a reason (racing condition render it useless).